common/djangoapps/third_party_auth/views.py

```diff
@@ -110,3 +112,14 @@ def post_to_custom_auth_form(request):
         'hmac': pipeline_data['hmac'],
     }
     return render(request, 'third_party_auth/post_custom_auth_entry.html', data)
+
+
+class IdPRedirectView(View):
```

Comment:
I decided not to extend Django's base view `RedirectView` because it returns a `410` `HttpResponseGone` when there is no url, and I wanted to return a `404`.

Reply:
pylint didn't complain about missing docstrings?

Reply:
I guess not. Weird. Thanks for catching that. I'll add those in tomorrow.

Reply:
I added docstrings. How do they look to you?